doc/reference/usermode/syscalls.rst

```diff
@@ -174,11 +174,13 @@ prototype for the implementation function is also automatically generated.
 
 The final layer is the invocation of the system call itself. All architectures
 implementing system calls must implement the seven inline functions
-:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`.  These
+:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`. These
 functions marshal arguments into designated CPU registers and perform the
-necessary privilege elevation. In this layer, all arguments are treated as an
-unsigned 32-bit type. There is always a 32-bit unsigned return value, which
-may or may not be used.
+necessary privilege elevation. Parameters of API inline function, before being
+passed as arguments to system call, are C casted to unsigned 32-bit integers,
+with exception to 64-bit parameters which are split into lower and higher
+32-bit unsigned int and passed as two consecutive arguments.
+There is always a 32-bit unsigned return value, which may or may not be used.
```

Comment:
@andrewboie  What about this? I have looked into generated code and the `z_mrsh` is returning `uintptr_t`, which is cast from whatever the `z_impl_` returns. Should this also be changed to "register size"?

Reply:
yeah any language that pins stuff to 32bits in size is all now uintptr_t and we always use a data model where that matches the register width

Reply:
Done.